apps/src/code-studio/components/ShareDialog.jsx

```diff
@@ -37,26 +47,137 @@ var ShareDialog = React.createClass({
     this.setState({isOpen: false});
   },
 
+  showSendToPhone: function (event) {
+    this.setState({
+      showSendToPhone: true,
+      showAdvancedOptions: false,
+    });
+    event.preventDefault();
+  },
+
+  showAdvancedOptions() {
+    this.setState({
+      showSendToPhone: false,
+      showAdvancedOptions: true,
+    });
+  },
+
+  clickExport: function () {
+    this.setState({exporting: true});
+    this.props.onClickExport().then(
+      this.setState.bind(this, {exporting: false}),
+      function () {
+        this.setState({
+          exporting: false,
+          exportError: 'Failed to export project. Please try again later.'
+        });
+      }.bind(this)
+    );
+  },
+
   render: function () {
+    var image;
+    var modalClass = 'modal-content';
+    if (this.props.icon) {
+      image = <img className="modal-image" src={this.props.icon}/>;
+    } else {
+      modalClass += ' no-modal-icon';
+    }
+
+    var facebookShareUrl = "https://www.facebook.com/sharer/sharer.php?u=" + this.props.encodedShareUrl;
+    var twitterShareUrl = "https://twitter.com/intent/tweet?url=" + this.props.encodedShareUrl +
+                          "&amp;text=Check%20out%20what%20I%20made%20@codeorg&amp;hashtags=HourOfCode&amp;related=codeorg";
+
+    var abuseStyle = {
+      border: '1px solid',
+      borderRadius: 10,
+      padding: 10,
+      marginBottom: 20
+    };
+
+    var abuseTextStyle = {
+      color: '#b94a48',
+      fontSize: 14
+    };
+
+    var abuseContents;
+    if (this.props.isAbusive) {
+      abuseContents = <AbuseError
+                          i18n={{
+                              tos: this.props.abuseTos,
+                              contact_us: this.props.abuseContact
+                            }}
+                          className='alert-error'
+                          style={abuseStyle}
+                          textStyle={abuseTextStyle}/>;
+    }
+
+    var sendToPhone;
+    if (this.state.showSendToPhone) {
+      sendToPhone = <SendToPhone
+                        channelId={this.props.channelId}
+                        appType={this.props.appType}
+                        styles={{label:{marginTop: 15, marginBottom: 0}}}
+                    />;
+    }
+
+    var advancedOptions;
+    if (this.props.appType === 'applab') {
+      advancedOptions = (
+        <AdvancedShareOptions
+            i18n={this.props.i18n}
+            onClickExport={this.props.onClickExport}
+            expanded={this.state.showAdvancedOptions}
+            onExpand={this.showAdvancedOptions}
+        />
+      );
+    }
```

Comment:
If we wanted to make these better match our style recommendations, we could move them into the return statement, instead of sticking them in variables, 
i.e. in our return something like:
```
{this.props.appType === 'applab' &&
  <AdvancedShareOptions .../>
}